insights/parsers/saphostctrl.py

```diff
@@ -0,0 +1,121 @@
+"""
+saphostctrl - Commands ``saphostctrl``
+======================================
+"""
+from .. import parser, CommandParser
+from ..parsers import ParseException
+from insights.specs import Specs
+from insights.core.filters import add_filter
+
+
+SAP_INST_FILTERS = [
+        '******',
+        'CreationClassName',
+        'SID',
+        'SystemNumber',
+        'InstanceName',
+        'Hostname',
+        'SapVersionInfo',
+]
+add_filter(Specs.saphostctl_getcimobject_sapinstance, SAP_INST_FILTERS)
+
+
+@parser(Specs.saphostctl_getcimobject_sapinstance)
+class SAPHostCtrlInstances(CommandParser):
+    """
+    This class provides processing for the output of the
+    ``/usr/sap/hostctrl/exe/saphostctrl -function GetCIMObject -enuminstances SAPInstance``
+    command on SAP systems.
+
+    Sample output of the command::
+
+        *********************************************************
+         CreationClassName , String , SAPInstance
+         SID , String , D89
+         SystemNumber , String , 88
+         InstanceName , String , HDB88
+         Hostname , String , hdb88
+         FullQualifiedHostname , String , hdb88.example.com
+         IPAddress , String , 10.0.0.88
+         SapVersionInfo , String , 749, patch 211, changelist 1754007
+        *********************************************************
+         CreationClassName , String , SAPInstance
+         SID , String , D90
+         SystemNumber , String , 90
+         InstanceName , String , HDB90
+         Hostname , String , hdb90
+         FullQualifiedHostname , String , hdb90.example.com
+         IPAddress , String , 10.0.0.90
+         SapVersionInfo , String , 749, patch 211, changelist 1754007
+
+    Examples:
+        >>> type(sap_inst)
+        <class 'insights.parsers.saphostctrl.SAPHostCtrlInstances'>
+        >>> sap_inst.data[-1]['CreationClassName']
+        'SAPInstance'
+        >>> sap_inst.data[-1]['SID']
+        'D90'
+        >>> sap_inst.data[-1]['SapVersionInfo']  # Note: captured as one string
+        '749, patch 211, changelist 1754007'
+        >>> sap_inst.data[0]['InstanceType']  # Inferred code from InstanceName
+        'HDB'
+
+    Attributes:
+        instances (list): The list of instances found in the cluster output.
+        sids (list): The list of SID found in the cluster output.
+        types (list): The list of instance types found in the cluster output.
```

Comment:
@xiangce you should also define the `data` attribute since that is what you show in the examples.

Reply:
@bfahr , Added as per your suggestion, Thanks.